ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/EarliestByOffsetKudaf.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.offset;
+
+import static io.confluent.ksql.function.udaf.offset.KudafByOffsetUtils.INTERMEDIATE_STRUCT_COMPARATOR;
+import static io.confluent.ksql.function.udaf.offset.KudafByOffsetUtils.VAL_FIELD;
+
+import io.confluent.ksql.GenericKey;
+import io.confluent.ksql.function.AggregateFunctionInitArguments;
+import io.confluent.ksql.function.BaseAggregateFunction;
+import io.confluent.ksql.function.ParameterInfo;
+import io.confluent.ksql.function.types.ParamTypes;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import java.util.Collections;
+import java.util.function.Function;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.streams.kstream.Merger;
+
+public class EarliestByOffsetKudaf extends BaseAggregateFunction<Object, Struct, Object> {
+
+  private final boolean ignoreNulls;
+  private final Schema internalStructSchema;
+
+  public EarliestByOffsetKudaf(
+      final AggregateFunctionInitArguments initArgs,
+      final SqlType aggregateType,
+      final SqlType returnType,
+      final boolean ignoreNulls,
+      final Schema internalStructSchema) {
+    super(EarliestByOffsetFactory.FUNCTION_NAME,
+        initArgs.udafIndex(),
+        () -> null,
```

Comment:
For the sake of readability, should we use constant like `Object INITIALIZED_VALUE = null` and use it here? When I see nulls, I always look at the function to see this is the value we want. I do see this is just initialization.

Reply:
I'm feeling cheap and want to suggest using a comment instead of having a separate object.  Would that be ok?